src/backend/distributed/deparser/deparse_seclabel_stmts.c

```diff
@@ -0,0 +1,78 @@
+/*-------------------------------------------------------------------------
+ *
+ * deparse_seclabel_stmts.c
+ *	  All routines to deparse SECURITY LABEL statements.
+ *
+ * Copyright (c), Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "distributed/deparser.h"
+#include "nodes/parsenodes.h"
+#include "utils/builtins.h"
+
+static void AppendSecLabelStmt(StringInfo buf, SecLabelStmt *stmt);
+
+/*
+ * DeparseSecLabelStmt builds and returns a string representing of the
+ * SecLabelStmt for application on a remote server.
+ */
+char *
+DeparseSecLabelStmt(Node *node)
+{
+	SecLabelStmt *secLabelStmt = castNode(SecLabelStmt, node);
+	StringInfoData buf = { 0 };
+	initStringInfo(&buf);
+
+	AppendSecLabelStmt(&buf, secLabelStmt);
+
+	return buf.data;
+}
+
+
+/*
+ * AppendSecLabelStmt generates the string representation of the
+ * SecLabelStmt and appends it to the buffer.
+ */
+static void
+AppendSecLabelStmt(StringInfo buf, SecLabelStmt *stmt)
+{
+	appendStringInfoString(buf, "SECURITY LABEL ");
+
+	if (stmt->provider != NULL)
+	{
+		appendStringInfo(buf, "FOR %s ", stmt->provider);
+	}
+
+	appendStringInfoString(buf, "ON ");
+
+	switch (stmt->objtype)
+	{
+		case OBJECT_ROLE:
+		{
+			appendStringInfo(buf, "ROLE %s ", strVal(stmt->object));
```

Comment:
Does that need quotes?

Reply:
Same answer as in https://github.com/citusdata/citus/pull/7304#discussion_r1387633697